internal/tracing/handlers/handler.go

```diff
@@ -20,7 +21,7 @@ var (
 	errNoVolumeMountForSecret = fmt.Errorf("no volumemount found for secret")
 )
 
-func BuildOptions(ctx context.Context, k8s client.Client, mcAddon *addonapiv1alpha1.ManagedClusterAddOn, userWorkloads addon.TracesOptions) (manifests.Options, error) {
+func BuildOptions(ctx context.Context, k8s client.Client, mcAddon *addonapiv1alpha1.ManagedClusterAddOn, userWorkloads addon.TracesOptions, instrumentationEnabled bool) (manifests.Options, error) {
```

Comment:
Correct me if I'm wrong but you don't need to pass `instrumentationEnabled` here since `userWorkload` already has that information

Reply:
That's true